database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1542,164 +1542,381 @@ $$
+            WHERE application_id = NEW.application_id
+            AND user_id = rev_id;
+
+        -- Insert new
+        WITH reviewer_action AS (
+            SELECT reviewer_action
+                FROM single_application_detail(NEW.application_id, rev_id)
+            ) 
+        INSERT INTO public.application_reviewer_action
+            (user_id, application_id, reviewer_action)  
+        VALUES(
+            rev_id,
+            NEW.application_id,
+            (SELECT reviewer_action FROM reviewer_action)
+         )
+         ON CONFLICT (user_id, application_id)
+         DO UPDATE
+            SET reviewer_action = (SELECT reviewer_action FROM reviewer_action);
+    END LOOP;
+    
+    
+    -- ASSIGNERS
+    IF TG_OP = 'UPDATE' THEN
+        assigner_ids = ARRAY(
+            SELECT assigner_id FROM review_assignment
+            WHERE application_id = NEW.application_id
+            AND assigner_id IS NOT NULL
+        );
+    ELSE
+        CASE WHEN NEW.assigner_id IS NULL THEN
+                assigner_ids = ARRAY[];
+            ELSE
+                assigner_ids = ARRAY[NEW.assigner_id];
+            END CASE;
+    END IF;
+
+    FOREACH ass_id IN ARRAY assigner_ids LOOP
+        -- Remove existing
+        UPDATE public.application_reviewer_action
+            SET assigner_action = NULL
+            WHERE application_id = NEW.application_id
+            AND user_id = ass_id;
+
+        -- Insert new
+        WITH assigner_action AS (
+            SELECT assigner_action
+                FROM single_application_detail(NEW.application_id, ass_id)
+            ) 
+        INSERT INTO public.application_reviewer_action
+            (user_id, application_id, assigner_action)  
+        VALUES(
+            ass_id,
+            NEW.application_id,
+            (SELECT assigner_action FROM assigner_action)
+         )
+         ON CONFLICT (user_id, application_id)
+         DO UPDATE
+            SET assigner_action = (SELECT assigner_action FROM assigner_action);
+    END LOOP;
+
+    -- Clean up NULL records
+    DELETE FROM public.application_reviewer_action
+    WHERE application_id = NEW.application_id
+    AND reviewer_action IS NULL
+    AND assigner_action IS NULL;
+
+RETURN NULL;
+
+END;
+$$
+LANGUAGE plpgsql;
+
+
+-- Trigger for the above on review_assignment
+DROP TRIGGER IF EXISTS update_application_reviewer_stats ON public.review_assignment;
+CREATE TRIGGER update_application_reviewer_stats
+    AFTER INSERT OR UPDATE ON public.review_assignment
+    FOR EACH ROW
+    EXECUTE FUNCTION public.update_application_reviewer_stats ();
+
+-- Function to update insert/update reviewer/assigner actions on
+-- application_reviewer_action when review_status is updated, by making calls to
+-- the above `single_application_detail` function for the appropriate
+-- user/application combo(s)
+DROP FUNCTION IF EXISTS public.update_reviewer_stats_from_status CASCADE;
+CREATE OR REPLACE FUNCTION public.update_reviewer_stats_from_status ()
```

Comment:
There's a bit more work for this one, as the relevant ids (application, reviewer, assigner) are not stored in the review_status_history table, so we fetch them first. After that, it's basically the same as the earlier previous function, except:
- no need to update the reviewer/assigner lists, they won't change here
- only need to update the single reviewer, so don't loop over other connected reviewers like this previous one

There's a bit of code duplication here, but I thought abstracting into new functions would make it more complicated than it needs to be -- unnecessary DRY ;)